### DISCUSSION TITLE
Project 3, Part 3

### DISCUSSION THREAD

**Discussion by Aidan-Myatt** [cooperrc/computational-mechanics#52] [Q&A]
Does anyone know what is wrong with my code here, and why incsearch is not working? I tried changing ns=3, but that gave me a different error saying invalid index to scalar variable.

![image](https://user-images.githubusercontent.com/106316906/174061957-0cccaa9f-e8af-448c-b470-4d514b0776d1.png)
![image](https://user-images.githubusercontent.com/106316906/174061506-086a6e3f-c9b0-44e4-abbf-256cd00a3ac3.png)
![image](https://user-images.githubusercontent.com/106316906/174061705-ea8832d7-4cb4-4215-ac45-3932d5168afe.png)

**Comment by cooperrc** [comment]
Take a look at how `incsearch` creates the `x` and `f(x)` 

```python
x = np.linpace(xmin, xmax, ns)
f = func(x)
```

The method assumes that you can pass an array into your function `func(x)`, but that's not always the case e.g. `func = lambda x: math.cos(x)` only accepts inputs of length 1. 

The workaround is [list comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)

```python
x = np.linpace(xmin, xmax, ns)
f = np.array([func(xi) for xi in x])
```
You evaluate one `xi` at a time in a loop, then save all the results as an array.

**Reply by Aidan-Myatt** [reply]
Thank you, that worked! Do you know why modified secant is not working here? I have been playing around with the input values, but nothing has worked.
![image](https://user-images.githubusercontent.com/106316906/174289794-36eb2fb8-5daa-4c59-9447-365789a41fc1.png)

**Reply by cooperrc** [reply]
If you get a `divide by zero` error, then your `dfunc` is $\approx 0$. 

Are you solving the `f_dm` function from the project? `x0=60` is very high. The fuel will be burned up in $t = \frac{0.2 kg}{60 kg/s} = 0.003 s$, not much time to lift off the ground.

**Reply by Aidan-Myatt** [reply]
The numbers I playing around with were too high. I changed it to 0.05, and the code is working. Thank you!
